disabled-tests-inspector/src/main/java/io/quarkus/ts/DisabledTestAnalyserService.java

```diff
@@ -0,0 +1,217 @@
+    );
+
+    private static final Pattern ISSUE_PATTERN = Pattern.compile(
+    "(https://(?:github\\.com/.+?/issues/\\d+|issues\\.redhat\\.com/browse/\\w+-\\d+))"
+    );
+
+
+    public void analyzeRepository(String repoOwner, String repoName, List<String> branches, String baseOutputFileName) throws IOException {
+        GitHub github = GitHub.connect();
+        GHRepository repo = github.getRepository(repoOwner + "/" + repoName);
+
+        for (String branch : branches) {
+            GHTree tree = repo.getTreeRecursive(branch, 1);
+            List<DisabledTest> disabledTests = new ArrayList<>();
+            Map<String, DisabledTestsModuleStats> moduleStats = new HashMap<>();
+
+            for (GHTreeEntry entry : tree.getTree()) {
+                String filePath = entry.getPath();
+                if (entry.getPath().contains("/test/") && filePath.endsWith(".java")) {
+                    GHContent fileContent = repo.getFileContent(filePath, branch);
+                    disabledTests.addAll(extractDisabledTests(fileContent, moduleStats));
+                }
+            }
+
+            BranchAnalysisResult result = new BranchAnalysisResult(branch, disabledTests);
+            String testListFile = getStatsOrTestFileName(baseOutputFileName, branch, false);
+            new ObjectMapper().writerWithDefaultPrettyPrinter().writeValue(new File(testListFile), result);
+
+            String statsListFile = getStatsOrTestFileName(baseOutputFileName, branch, true);
+            new ObjectMapper().writerWithDefaultPrettyPrinter().writeValue(new File(statsListFile), moduleStats);
+        }
+    }
+
+    private List<DisabledTest> extractDisabledTests(GHContent ghContent,
+                                                    Map<String, DisabledTestsModuleStats> moduleStats) throws IOException {
+
+        String fileUrl = ghContent.getHtmlUrl();
+        String fileContent = ghContent.getContent();
+
+        List<DisabledTest> disabledTests = new ArrayList<>();
+        String[] lines = fileContent.split("\n");
+
+        String previousLine = "";
+        String currentClass = "UnknownClass";
+        String currentTestMethod = null;
+        List<String> annotationTypes = new ArrayList<>();
+        List<String> reasons = new ArrayList<>();
+        List<String> issueLinks = new ArrayList<>();
+        boolean insideDisabledBlock = false;
+
+        for (String currentLine : lines) {
+            currentLine = currentLine.trim();
+
+            Matcher classMatcher = CLASS_DECLARATION_PATTERN.matcher(currentLine);
+            if (classMatcher.find()) {
+                currentClass = classMatcher.group(1);
+            }
+
+            Matcher testMethodMatcher = TEST_METHOD_PATTERN.matcher(currentLine);
+            if (testMethodMatcher.find()) {
+                currentTestMethod = testMethodMatcher.group(1);
+            }
+
+            Matcher disabledMatcher = DISABLED_ANNOTATION_PATTERN.matcher(currentLine);
```

Comment:
You can also disable tests by enabling them conditionally, e.g. `@EnabledOnQuarkusVersion(version = "1\\..*", reason = "Redirection is no longer supported in 2.x")`, `@EnabledIfSystemProperty(named = "ts.redhat.registry.enabled", matches = "true")`.